src/auto-evo/EvolutionaryTree.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoEvo;
+using Godot;
+using Godot.Collections;
+
+public class EvolutionaryTree : Control
+{
+    private const float LEFT_MARGIN = 5.0f;
+    private const float TIMELINE_HEIGHT = 50.0f;
+    private const float TIMELINE_LINE_THICKNESS = 2.0f;
+    private const float TIMELINE_LINE_Y = 5.0f;
+    private const float TIMELINE_MARK_SIZE = 4.0f;
+    private const float SPECIES_SEPARATION = 50.0f;
+    private const float GENERATION_SEPARATION = 100.0f;
+    private const float SPECIES_NAME_OFFSET = 10.0f;
+
+    private readonly List<EvolutionaryTreeNode> nodes = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, string> speciesNames = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, EvolutionaryTreeNode> latestNodes = new();
+
+    private readonly System.Collections.Generic.Dictionary<uint, (uint ParentSpeciesID, int SplitGeneration)>
+        speciesOrigin = new();
+
+    private readonly ButtonGroup nodesGroup = new();
+
+    private Font latoSmallItalic = null!;
+    private Font latoSmallRegular = null!;
+
+    private PackedScene treeNodeScene = null!;
+
+    private Vector2 treeNodeSize;
+
+    private uint maxSpeciesId;
+
+    private int latestGeneration;
+
+    [Signal]
+    public delegate void SpeciesSelected(int generation, uint id);
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        treeNodeScene = GD.Load<PackedScene>("res://src/auto-evo/EvolutionaryTreeNode.tscn");
+
+        latoSmallItalic = GD.Load<Font>("res://src/gui_common/fonts/Lato-Italic-Small.tres");
+        latoSmallRegular = GD.Load<Font>("res://src/gui_common/fonts/Lato-Regular-Small.tres");
+    }
+
+    public void Init(Species luca)
+    {
+        SetupTreeNode(luca, null, 0);
+        treeNodeSize = nodes[0].RectSize;
+
+        speciesOrigin.Add(luca.ID, (uint.MaxValue, 0));
+        speciesNames.Add(luca.ID, luca.FormattedName);
+    }
+
+    public override void _Draw()
+    {
+        base._Draw();
+
+        // Draw timeline
+        DrawLine(new Vector2(0, TIMELINE_LINE_Y), new Vector2(RectSize.x, TIMELINE_LINE_Y), Colors.Cyan,
+            TIMELINE_LINE_THICKNESS, true);
+
+        for (int i = 0; i <= latestGeneration; i++)
+        {
+            DrawLine(new Vector2(LEFT_MARGIN + i * GENERATION_SEPARATION + treeNodeSize.x / 2, TIMELINE_LINE_Y),
+                new Vector2(LEFT_MARGIN + i * GENERATION_SEPARATION + treeNodeSize.x / 2,
+                    TIMELINE_LINE_Y + TIMELINE_MARK_SIZE),
+                Colors.Cyan, TIMELINE_LINE_THICKNESS, true);
+
+            var localizedText = i + " " + TranslationServer.Translate("MEGA_YEARS");
```

Comment:
This line is still the wrong version, this is my last code related comment. Everything else has now been fixed.

Reply:
I'm not familiar with the number format you use in the code, so I cannot use it like that without doing some 'i * 1e6' thing.

If you know how to make it work correctly without multiplying 1e6, you may change that yourself. It wouldn't be a tough thing.

Reply:
It shows correctly in the editor, I'm pretty sure this format specifier removes the scientific notation:
```
Text = string.Format(CultureInfo.CurrentCulture, "{0:#,##0,,}", data.TimePeriod) + " "
                    + TranslationServer.Translate("MEGA_YEARS"),
```
`{0:#,##0,,}` is the part you need. I think @Kasterisk wrote that initially and can clarify.

Reply:
Anyway.

Reply:
Hang on now that I read about this, you are using `i` as the generation number? That doesn't map to our years fully. I think maybe you should just use `i.ToString(CultureInfo.CurrentCulture)` without any suffix or something like that? If you add a suffix it should be like "G" for "generation" as `i` is the generation number and not a year as I thought before.

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-numeric-format-strings

Ah, generation i is 100 * i Myr

Reply:
Currently, but not necessarily forever, changing the game world time elapse code will happen at some point and then the auto-evo tool will once again show incorrect values.

Reply:
Note that evolutionary tree will be a main game thing, so the world time will eventually be applied.

Reply:
I guess yeah. Well the solution where the 0 is printed as "000 Myr" is not very nice looking to me.

Reply:
That's purely accident...